### PR TITLE
feat(policy): Add `MeshMetric` api

### PR DESCRIPTION
**Pull request by Automaat**
Fixes: #8575

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/plugins/policies/meshmetric/plugin/v1alpha1/plugin.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+func NewPlugin() core_plugins.Plugin {
+	return &plugin{}
+}
+
+func (p plugin) MatchedPolicies(dataplane *core_mesh.DataplaneResource, resources xds_context.Resources) (core_xds.TypedMatchingPolicies, error) {
+	return matchers.MatchedPolicies(api.MeshMetricType, dataplane, resources)
+}
+
+func (p plugin) Apply(rs *core_xds.ResourceSet, ctx xds_context.Context, proxy *core_xds.Proxy) error {
+	log.Info("apply is not implemented")
```
Comment (slonka) [inline pkg/plugins/policies/meshmetric/plugin/v1alpha1/plugin.go:29]: nitpick: maybe change it to debug until we implement this not to spam logs, or just remove altogether

**Review by slonka — COMMENTED** [review]
some validation missing, otherwise 👍

File: pkg/plugins/policies/meshmetric/api/v1alpha1/validator.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+func validateSidecar(sidecar *Sidecar) validators.ValidationError {
+	var verr validators.ValidationError
+	if sidecar == nil {
+		return verr
+	}
+
+	if sidecar.Regex != nil {
+		_, err := regexp.Compile(*sidecar.Regex)
```
[the remark this answers is not in the document]
Comment (Automaat) [reply pkg/plugins/policies/meshmetric/api/v1alpha1/validator.go:42]: yes, this is the same engine: https://github.com/google/re2

File: pkg/plugins/policies/meshmetric/plugin/v1alpha1/plugin.go (hunk 1 above)
Comment (Automaat) [reply pkg/plugins/policies/meshmetric/plugin/v1alpha1/plugin.go:29]: but, you won't be applying this nowhere until we implement it, so I don't think we need to change it now

File: pkg/plugins/policies/meshmetric/api/v1alpha1/validator.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	if backends == nil {
+		return verr
+	}
+
+	for idx, backend := range *backends {
+		path := validators.RootedAt("backend").Index(idx)
+		switch backend.Type {
+		case PrometheusBackendType:
+			if backend.Prometheus == nil {
+				verr.AddViolationAt(path.Field("prometheus"), validators.MustBeDefined)
```
[the remark this answers is not in the document]
Comment (Automaat) [reply pkg/plugins/policies/meshmetric/api/v1alpha1/validator.go:62]: do we have some common `path` validator? 

Also, does tls field need any validation?

File: pkg/plugins/policies/meshmetric/plugin/v1alpha1/plugin.go (hunk 1 above)
Comment (slonka) [reply pkg/plugins/policies/meshmetric/plugin/v1alpha1/plugin.go:29]: i think it's called anyway it just does not match any DP so it's a noop, can stay but it's a bit annoying

Comment (Automaat) [reply pkg/plugins/policies/meshmetric/plugin/v1alpha1/plugin.go]: changed

File: pkg/plugins/policies/meshmetric/api/v1alpha1/validator.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+			common_api.Mesh, common_api.MeshSubset, common_api.MeshService, common_api.MeshServiceSubset,
+		},
+	})
+	return targetRefErr
+}
+
+func validateDefault(conf Conf) validators.ValidationError {
+	var verr validators.ValidationError
+	verr.AddError("sidecar", validateSidecar(conf.Sidecar))
+	verr.AddError("backends", validateBackend(conf.Backends))
```
[the remark this answers is not in the document]
Comment (Automaat) [reply pkg/plugins/policies/meshmetric/api/v1alpha1/validator.go]: added validation

**Review by slonka — APPROVED** [review]

**Pull request merged**